test/integration/targets/noexec/test-noexec.yml

```diff
@@ -0,0 +1,8 @@
+- hosts: localhost
+  gather_facts: false
+  tasks:
+    - ping:
```

Comment:
Why is this ping needed?

Reply:
To validate the difference in calling with and without async.  The module is ultimately executed differently in the end, just a test case to ensure it continues working with `noexec`.

Reply:
To clarify, this PR deals with `async_wrapper`, but this specific task is effectively targeting this line, to ensure we build the module execution command in a way that still works with noexec:

https://github.com/ansible/ansible/blob/0d7c144ce44cd40ffa7c109a027d0927961d6a63/lib/ansible/plugins/action/__init__.py#L1078